app/person-escort-record/controllers/frameworks.js

```diff
@@ -19,6 +21,48 @@ class FrameworksController extends FormWizardController {
     next()
   }
 
+  reduceResponses(accumulator, { value, value_type: valueType, question }) {
+    const field = question.key
+
+    if (valueType === 'object') {
+      accumulator[field] = value.option
+      accumulator[`${field}--${kebabCase(value.option)}`] = value.details
+    }
+
+    if (valueType === 'collection') {
+      accumulator[field] = value.map(item => item.option)
+      value.forEach(item => {
+        accumulator[`${field}--${kebabCase(item.option)}`] = item.details
+      })
+    }
+
+    if (valueType === 'string' || valueType === 'array') {
+      accumulator[field] = value
+    }
+
+    return accumulator
+  }
+
+  getValues(req, res, callback) {
+    const fields = req.form.options.fields
+    const responses = req.personEscortRecord.responses
+    const savedValues = responses
+      .filter(response => fields[response.question.key])
+      .filter(response => response.value)
+      .reduce(this.reduceResponses, {})
```

Comment:
_This is also a very minor comment that can be easily ignored as the outcome doesn't change._

`reduceResponse` is just a pure function, there is no benefit on having it as a method on the class. You could have it as a single function and remove the need to `this.XXXX`. `this` in `reduce` statements is a bit tricky, and reading that lines sort of made me think... "does it uses class properties? why does it have to be a method on the class? is the scope relevant?"
A simple function would have not raised the same questions.

Reply:
That's a valid point. So the reason for exposing it on the class was mainly to make it easier to test in isolation. That meant I could mock it when testing this function and test the functionality in that method in isolation.

If it's added as a private method I'd need to then test that functionality within `getValues` which I felt makes reading those tests harder and involves more boilerplate code in the tests.

Any thoughts on how I could test it in isolation? Maybe I could move it to `helpers`...?

Reply:
I started to have a look at this but then it differs from what we've done elsewhere at the moment. I agree it would be nicer to not use a method on the class where we don't need access to the class properties but I think to avoid wasting too much time for now I'll leave it as it is.

Also, I'm aware I will need to extend this shortly once the endpoint is there to ensure values are processed correctly and sent in the same format so I will look to re-work it as part of that work.